pkg/util/pki/parse.go

```diff
@@ -252,8 +252,14 @@ func (c *chainNode) toBundleAndCA() (PEMBundle, error) {
 
 	for {
 		// If the issuer is nil, we have hit the root of the chain. Assign the CA
-		// to this certificate and stop traversing.
+		// to this certificate and stop traversing. If the certificate at the root
+		// of the chain is not self-signed (i.e. is not a root CA), then also append
+		// that certificate to the chain.
```

Comment:
Suggestion: would it be worth linking https://datatracker.ietf.org/doc/html/rfc5246#section-7.4.2 here, specifically the bit where 'the root certificate authority MAY be omitted from the chain' just so that it's clearer why root and only root is omitted?

Reply:
Yeah, good idea, I'll do that now :+1:

Reply:
Thanks!